core/src/main/java/com/netease/arctic/scan/TableEntriesScan.java

```diff
@@ -162,7 +180,10 @@ public CloseableIterable<IcebergFileEntry> entries() {
           if (shouldKeep(status, fileContent)) {
             ContentFile<?> contentFile = buildContentFile(fileContent, fileRecord);
             if (metricsEvaluator().eval(contentFile)) {
-              return new IcebergFileEntry(snapshotId, sequence, contentFile);
+              ContentFile<?> copyFile = includeColumnStats ?
```

Comment:
Why do need to copy the `contentFile` here?

Reply:
Before copy, the `contentFile` may contain column stats even if `includeColumnStats == false`, because of `dataFilter != null`.

```
private boolean needMetrics() {
    return dataFilter != null || includeColumnStats;
  }
```

So we call `contentFile.copyWithoutStats()` to remove these column stats.

However, it's not necessary to copy it in other cases.

Reply:
I modified the code, you may take a look.